### PR TITLE
[Enhancement] Open clang compile warning option and resolve compile error Part1

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

Issue Number: #8013

This pr remove those compile options:
```cmake
-Wno-address-of-packed-member
-Wno-ambiguous-reversed-operator
-Wno-atomic-implicit-seq-cst
-Wno-unused-but-set-variable
-Wno-unused-const-variable
-Wno-unused-exception-parameter
-Wno-unused-function
-Wno-unused-local-typedef
-Wno-unused-value
-Wno-unused-variable
-Wno-variadic-macros
-Wno-zero-as-null-pointer-constant
```

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/es/es_scroll_parser.cpp [hunk 1]
```diff
@@ -258,8 +256,7 @@ static Status insert_int_value(const rapidjson::Value& col, PrimitiveType type,
     return Status::OK();
 }
 
-ScrollParser::ScrollParser(bool doc_value_mode)
-        : _scroll_id(""), _size(0), _line_index(0), _doc_value_mode(doc_value_mode) {}
+ScrollParser::ScrollParser(bool doc_value_mode) : _scroll_id(""), _size(0), _line_index(0) {}
```
Comment (morningman) [inline be/src/exec/es/es_scroll_parser.cpp:259]: `doc_value_mode` is not used?

Comment (BiteTheDDDDt) [reply be/src/exec/es/es_scroll_parser.cpp:259]: > `doc_value_mode` is not used?

yes

Comment (morningman) [reply be/src/exec/es/es_scroll_parser.cpp:259]: So why not remove it?

Comment (BiteTheDDDDt) [reply be/src/exec/es/es_scroll_parser.cpp]: > So why not remove it?

Now I have remove it.

**Comment by yangzhg** [comment]
I had a pr #8069  have fix all warnings

**Pull request closed without merging**